Model/TransactionRepository.php

```diff
@@ -194,6 +229,187 @@ public function save(
     /**
      * {@inheritdoc}
      */
+    public function processTransaction(
+        $cartId,
+        $transactionId,
+        \Magento\Quote\Api\Data\PaymentInterface $paymentMethod,
+        \Gr4vy\Magento\Api\Data\MethodInterface $methodData,
+        \Gr4vy\Magento\Api\Data\ServiceInterface $serviceData,
+        \Gr4vy\Magento\Api\Data\TransactionInterface $transactionData
+    ) {
+        $transaction = $this->transactionApi->getTransactionDetail($transactionId);
+        $this->parseGr4vyRawData($paymentMethod, $methodData, $serviceData, $transactionData, $transaction);
+        $this->setPaymentInformation($cartId, $paymentMethod, $methodData, $serviceData, $transactionData);
+    
+        $statuses = $this->orderHelper->getGr4vyTransactionStatuses();
+
+        if (in_array($transactionData->getStatus(), $statuses['cancel'])) {
+            $result = $this->saveFailedOrder($cartId, $paymentMethod, $methodData, $serviceData, $transactionData);
+            return json_encode($result);
+        } else if (
+            in_array($transactionData->getStatus(), $statuses['processing']) ||
+            in_array($transactionData->getStatus(), $statuses['success']) 
+        ){
+            $result = [];
+            $result['success'] = true;
+            return json_encode($result);
+        }
+    
+        $result = [];
+        $result['success'] = false;
+        return json_encode($result);
+    }
+
+    /**
+     * Set the PaymentMethod Data with the data received from Gr4vy
+     *
+     * @param \Magento\Quote\Api\Data\PaymentInterface paymentMethod
+     * @param array rawData
+     */
+
+    private function parseGr4vyRawData($paymentMethod, $methodData, $serviceData, $transactionData, $transactionRawData) {
```

Comment:
Minor, since we're using PHPDoc comments

```suggestion
    /**
     * Set the PaymentMethod Data with the data received from Gr4vy
     *
     * @param \Magento\Quote\Api\Data\PaymentInterface $paymentMethod
     * @param \Gr4vy\Magento\Api\Data\MethodInterface $methodData
     * @param \Gr4vy\Magento\Api\Data\ServiceInterface $serviceData
     * @param \Gr4vy\Magento\Api\Data\TransactionInterface $transactionData
     * @param object $transactionRawData
     */

    private function parseGr4vyRawData($paymentMethod, $methodData, $serviceData, $transactionData, $transactionRawData) {
```